### DISCUSSION TITLE
Book Shelves

### DISCUSSION THREAD

**Discussion by shirish87** [PaulWoitaschek/Voice#1214] [Ideas]
I'm curious if people are interested in having features for organizing larger libraries. These may be fringe use-cases.

## Feature 1: Mark an audiobook as "Current" or "Not Started Yet" or "Completed"

- Tabbed UI 
- This is useful when you copy books you've previously finished to your phone and wish the keep the "Current" queue clean
- Or sometimes you start a book but totally lose context and want to re-read it later
- An alternative is to have 2 tabs: "Current" and "Library". Library tab would later have filters for Completed / never read.

## Feature 2: User-defined book shelves (or labels/tags)

- Current option to edit the book name is helpful, but people seem to want to edit other metadata as well (authors, series, etc.)
- A book-shelf here is just a generic label or tag that can be set for a book (e.g. favourite author, book series name, etc)
- An audiobook can belong to multiple bookshelves
- Lists can be filtered using these

https://user-images.githubusercontent.com/304687/165751157-87121152-adf8-466d-9835-1111dc9fdef7.mp4

**Comment by mudpriest** [comment]
I think this is a great Idea!

**Comment by microraptor** [comment]
It would be awesome, if during the setup process of adding a folder to Voice and selecting the folder structure, there would be also a toggle to have the first level sub-folders (parent folders of authors/books) determine the book shelf described in feature 2.

So the book shelves are determined by the folder structure like this:

- Audiobooks/
  - Fiction/
    - Title of Fiction Book 1
    - Title of Fiction Book 2
  - Non-Fiction/
    - Title of Non-Fiction Book 1
    - Title of Non-Fiction Book 2
  - Horror/
    - Title of Horror Book

Just to be clear, this would be an optional toggle during the process of adding a folder.

**Comment by MNeMoNiCuZ** [comment]
Yes! Something like this would be great!
I sorely miss some way of arranging a large collection of books.
I would love some automatic (ideally), genre/category sorting.

**Comment by egvimo** [comment]
It would be great to be able to organize large book collections. Either this way or any other of the ideas discussed here.

Maybe just using the ID3 metadata for the categories would provide a good starting point.

**Comment by shiiverstorm** [comment]
I would love this too! commenting to follow :)

**Comment by JP-Garcia** [comment]
I have my own idea, which probably isn't as good but I would hope it would be easier / simpler to implement.

Current behaivior: Each folder is a single book, and each file is a chapter.

Desired behaivior: Option to treat a folder as a collapsable list of books, or even a list inside a submenu. Ideally this would be a toggle in the book's long-press menu.

Use case: I have over 100 audiobooks now that are impossible to scroll through. I would like to organize some of them by author or series, and especially to group together short stories under their single author. I can technically already do this by sticking them all in their own folder. However 1) they lose their individual cover images, which has been a great aid in picking out the next book to listen to, and 2) it does not work well for books already split by chapter.

I was going to post this as a new idea but I think it belongs here.
